### PR TITLE
Add CRUD functionality for Shelter Reviews

### PR DESCRIPTION
**Pull request by Shock253**
Add pages for creating, updating and deleting reviews to be viewed on the shelter show page

### CODE REVIEW

File: app/controllers/shelter_reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class ShelterReviewsController < ApplicationController
+  def edit
+    @shelter =  Shelter.find(params[:id])
+    @review = @shelter.shelter_reviews.find(params[:review_id])
```
Comment (zachholcomb) [inline app/controllers/shelter_reviews_controller.rb]: good catch on getting reviews from shelters

File: app/controllers/shelter_reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def update
+    shelter = Shelter.find(params[:id])
+    review = ShelterReview.find(params[:review_id])
+    review.update(shelter_review_params)
+    if review.save
+      redirect_to "/shelters/#{shelter.id}"
+    else
+      flash[:notice] = "Update not recorded: Please enter title, rating, and content to edit review."
```
Comment (zachholcomb) [inline app/controllers/shelter_reviews_controller.rb]: we'll have to rename flash notices to be unique right?

File: Gemfile.lock [hunk 3]
```diff
@@ -212,5 +212,8 @@ DEPENDENCIES
   uglifier (>= 1.3.0)
   web-console (>= 3.3.0)
 
+RUBY VERSION
+   ruby 2.5.3p105
+
```
Comment (Shock253) [inline Gemfile.lock]: We're using the version 2.5.3 up on heroku, I'm not sure if this affects the local repos, but I think it's best to be operating on the same version of ruby anyways

File: app/models/shelter.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class Shelter < ApplicationRecord
   has_many :pets, dependent: :destroy
+  has_many :shelter_reviews
```
Comment (zachholcomb) [inline app/models/shelter.rb]: we might need a dependent destroy clause here, not sure though

File: app/views/shelter_reviews/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<ul class="nav">
```
Comment (zachholcomb) [inline app/views/shelter_reviews/edit.html.erb]: in the next branch we will move the nav bar to application layout

File: spec/features/shelter_reviews/shelter_reviews_create_spec.rb [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    fill_in "content", with: ""
+
+    click_button "Post Review"
+
+    expect(page).to have_content("Could not create shelter: Please make sure to enter title, rating, and content")
+    expect(page).to have_current_path("/shelters/#{shelter_1.id}/reviews/new")
+  end
+end
+
+# User Story 4, Shelter Review Creation, cont.
```
Comment (zachholcomb) [inline spec/features/shelter_reviews/shelter_reviews_create_spec.rb]: looks like we forgot to remove this, we should clean up next pull request

File: app/models/shelter.rb [hunk 7]
```diff
@@ -1,5 +1,6 @@
...
   validates_presence_of :name, :address, :city, :state, :zip
 
   def pet_count
```
[the remark this answers is not in the document]
Comment (zachholcomb) [reply app/models/shelter.rb]: cool added dependent destroy we should be able to delete shelters with reviews now

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -9,6 +9,11 @@
   </head>
 
   <body>
```
[the remark this answers is not in the document]
Comment (zachholcomb) [reply app/views/layouts/application.html.erb]: yeah i agree, we'll have to add this to the next pull request

File: spec/features/shelter_reviews/shelter_reviews_create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    shelter_1 = Shelter.create!(name: "Denver Animal Shelter",
+                               address: "500 Invisible St.",
+                               city: "Denver",
+                               state: "Colorado",
+                               zip: "80201")
+
+    visit "/shelters/#{shelter_1.id}/reviews/new"
+
+    fill_in "title", with: ""
+    fill_in "rating", with: ""
```
Comment (zachholcomb) [inline spec/features/shelter_reviews/shelter_reviews_create_spec.rb]: nice testing of edge case!

**Pull request merged**